shell/test.sh

```diff
@@ -128,6 +128,11 @@ set -ex
 # Why: We want these to split. For those wondering about "$@":
 # https://stackoverflow.com/questions/5720194/how-do-i-pass-on-script-arguments-that-contain-quotes-spaces
 # shellcheck disable=SC2086
-go test $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
+"$DIR/gobin.sh" gotest.tools/gotestsum@v1.7.0 --junitfile bin/unit-tests.xml -- $BENCH_FLAGS $COVER_FLAGS $TEST_FLAGS \
```

Comment:
Rather than lock it to 1.7.0 explicitly mind putting this in an environment variable? Just makes it more configurable

Reply:
We don't really use environment variables (and we don't want to start doing that either).  

I think we should use the service.yaml versions override setup for this like we do for other tools.

Reply:
Fixed